### PR TITLE
Add 'repo ops' back into events

### PR DESCRIPTION
**Pull request by whyrusleeping**
This isnt ready to merge yet, still waiting on the upstream lexicon updates, but it *is* backwards compatible with the existing lexicons so it works against staging still

### CODE REVIEW

File: cmd/gosky/main.go [hunk 1]
```diff
@@ -823,6 +824,29 @@ var resetPasswordCmd = &cli.Command{
 	},
 }
 
+var updateHandleCmd = &cli.Command{
```
Comment (bnewbold) [inline cmd/gosky/main.go:827]: It would be nice to pull out commits like this into separate PRs. And the `--auth` revert to a seperate commit.

File: indexer/keymgr.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func (km *KeyManager) getKey(ctx context.Context, did string) (*did.PubKey, erro
 		return nil, err
 	}
 
-	pubk, err := doc.GetPublicKey("#signingKey")
```
Comment (bnewbold) [inline indexer/keymgr.go:51]: Another (important!) fix that would be good in a separate PR or at least commit

File: bgs/bgs.go [hunk 3]
```diff
@@ -116,10 +124,13 @@ func (bgs *BGS) HandleHealthCheck(c echo.Context) error {
...
-	Handle string `gorm:"uniqueIndex"`
-	Did    string `gorm:"uniqueIndex"`
-	PDS    uint
+	ID        bsutil.Uid `gorm:"primarykey"`
+	CreatedAt time.Time
+	UpdatedAt time.Time
+	DeletedAt gorm.DeletedAt `gorm:"index"`
+	Handle    string         `gorm:"uniqueIndex"`
+	Did       string         `gorm:"uniqueIndex"`
+	PDS       uint
```
Comment (bnewbold) [inline bgs/bgs.go:133]: Will the gorm auto-migrate code handle this sort of non-trivial migration? Or will we need to wipe state in staging to deploy this (`DROP DATABASE` and rebuild)?

As we move to having stuff deployed we should both add "deployment concerns" sections or comments to PR messages, so it is clear how and when they can go out.

File: cmd/gosky/util/util.go [hunk 4]
```diff
@@ -152,6 +152,8 @@ func SetupDatabase(dburl string) (*gorm.DB, error) {
 	var dial gorm.Dialector
 	// NOTE(bnewbold): might also handle file:// as sqlite, but let's keep it
 	// explicit for now
+
+	openConns := 99
```
Comment (bnewbold) [inline cmd/gosky/util/util.go:156]: I think this is pretty high for postgresql by default? I know there have been improvements in recent years, but I think the postgresql design philosophy is to keep connections to a couple dozen at max and go with `pg_bounce` if needed. Probably fine on our beefy deployed instances. I would have gone with 20 or 30.

File: util/uid.go [hunk 5]
```diff
@@ -0,0 +1,3 @@
+package util
+
+type Uid uint
```
Comment (bnewbold) [inline util/uid.go:3]: :+1: big fan of this sort of thing

**Review by bnewbold — APPROVED** [review]
Looks reasonable over all. I am curious what you think of that gorm migration: can we just deploy this as-is to staging?

My comments here are mostly opinions (about postgresql connection counts and git workflow).

File: util/uid.go (hunk 5 above)
Comment (whyrusleeping) [reply util/uid.go:3]: Yeah, it makes messing things up a lot harder lol

File: cmd/gosky/util/util.go (hunk 4 above)
Comment (whyrusleeping) [reply cmd/gosky/util/util.go:156]: this is really just copy/paste from my estuary code, picked it there on recommendation of some perf guides somewhere, totally not opinionated about it

File: bgs/bgs.go (hunk 3 above)
Comment (whyrusleeping) [reply bgs/bgs.go:133]: yeah this is actually not even a migration, its all the same fields, just one of them is getting a named type instead of a bare `uint`

File: cmd/gosky/main.go (hunk 1 above)
Comment (whyrusleeping) [reply cmd/gosky/main.go:827]: Yeah, i'm moving towards that, but things are still in the phase of needing a bunch of things to land all at the same time to stay up to date.

File: bgs/bgs.go (hunk 3 above)
Comment (bnewbold) [reply bgs/bgs.go:133]: The `ID` field is new, and the primary key; will it be auto-generated on existing rows in the database?

Also all the timestamp fields are new, and don't seem to be nullable. Will they default to "now"?
